packages/knip/src/plugins/eleventy/index.ts

```diff
@@ -32,19 +33,39 @@ const findEleventyDependencies: GenericPluginCallback = async (configFilePath, o
     return config.entry
       ? config.entry.map(toProductionEntryPattern)
       : PRODUCTION_ENTRY_FILE_PATTERNS.map(toProductionEntryPattern);
-  if (typeof localConfig === 'function') localConfig = await localConfig(new DummyEleventyConfig());
+
+  const dummyUserConfig = new DummyEleventyConfig();
+  if (typeof localConfig === 'function') localConfig = await localConfig(dummyUserConfig);
 
   const inputDir = localConfig?.dir?.input || defaultEleventyConfig.dir.input;
   const dataDir = localConfig?.dir?.data || defaultEleventyConfig.dir.data;
   const templateFormats = localConfig.templateFormats || defaultEleventyConfig.templateFormats;
 
-  return (
-    config?.entry ?? [
-      join(inputDir, dataDir, '**/*.js'),
-      join(inputDir, `**/*.{${typeof templateFormats === 'string' ? templateFormats : templateFormats.join(',')}}`),
-      join(inputDir, '**/*.11tydata.js'),
-    ]
-  ).map(toEntryPattern);
+  const copiedPackages = [];
+  const copiedEntries = [];
+
+  for (const path of Object.keys(dummyUserConfig.passthroughCopies).map(normalize)) {
```

Comment:
Normalized so we don't have to deal with `./`, `/`, etc.

Reply:
I'm going to use this particular comment to reply, but it covers a few of the following lines as well. It's a combination of things why I think we can improve this. little:

- paths including `node_modules` don't need to be translated to a package name here, Knip core has ways to deal with that (e.g. `@org/name` is also a valid package name)
- I've seen directories that end with `/` and some that don't
- The `DEFAULT_EXTENSIONS` already include the dot in e.g. `.js`

When fiddling with some cases I ended up with this:

```ts
const exts = DEFAULT_EXTENSIONS.map(extname => extname.slice(1)).join(',');
const copiedPackages = new Set<string>();
const copiedEntries = new Set<string>();

for (const path of Object.keys(dummyUserConfig.passthroughCopies)) {
  const isDir = !path.includes('*') && isDirectory(join(dirname(configFilePath), path));
  if (isDir) {
    copiedEntries.add(join(path, `**/*.{${exts}}`));
  } else if (isInternal(path)) {
    copiedEntries.add(path);
  } else {
    copiedPackages.add(path);
  }
}
```

Eager to learn what you think of this and if this works properly for your use cases well?